### ISSUE TITLE
Help on plugin

### ISSUE THREAD

**Issue by psukys** [eliostvs/tomate#16]
Hey, @eliostvs , can you help me on some issues with my plugin [1] (I know that this is likely not the place to ask, just that I couldn't find a better medium for this). I want to make it have automated builds and all that your other plugins have so that they can be installed through `apt`. So far I just have it working and the method of deployment is moving files into folders and activating through plugin window.

Even though the plugin works, there's a lot that can be done as well (to improve UX, etc.))

[1] - https://github.com/psukys/tomate-toggl-plugin

**Comment by eliostvs** [comment]
Hi, it seems great, congratulations! I'll definitely check. Nowadays I'm using [OpenSuse build service](https://build.opensuse.org/) to build and host the artifacts for Debian, Fedora, OpenSuse and Ubuntu. I've a [subproject](https://build.opensuse.org/project/show/home:eliostvs:tomate) with "jobs" for every tomate package (python-tomate, tomate-gtk, etc), they read the respective github master branch repository every time the job is trigger, so I don't need to change the job, to update the package version for example, once the configuration is done. The tests are running in Travis CI. The idea is every passing test in the master branch will trigger a package build in the openSuse build service. I've stopped using ppa years ago so I can't tell if you can do the same with it.

**Comment by psukys** [comment]
Ah, understood. Well OpenSuse PPA works as well. I'll read more about them in the weekend, if possible, I'll keep the issue open or should we move the discussion elsewhere?

**Comment by psukys** [comment]
@eliostvs a rather different question, but how does one use several files for the plugin?
That is, if I have a locally defined file which gets copied to `/usr/share/tomate/plugins` (or should it be put somewhere else?), what's the common way of using it in the path?

**Comment by eliostvs** [comment]
I think that you can use as many modules as you need as long they are accessible from your entry module, the one that is configured in the *.plugin file. 

You can put the plugins in the directories:  `~/.local/share/tomate/plugins:/usr/local/share/tomate/plugins:/usr/share/tomate/plugins`.  Normally I use `~/.local/share/` when I'm developing a plugin.

**Comment by psukys** [comment]
Thanks for the tips, found a solution for splitting up :)

Another question would be regarding general strategy on managing dependencies. Is everything done within the operating system's package manager or are dependencies resolved through `pip` as well?

**Comment by eliostvs** [comment]
Well, I'm using the distro package manager to handle the dependencies. I do think this is the best chosen. I do not want to handle updates/removes of dependencies or mess up the system by accident. In the other hand, Linux packing still hard but you can copy my repos in OBS and only change what you need.

**Comment by eliostvs** [comment]
If you still have questions, please open a new issue.
